certif/app/components/auth/login-form.js

```diff
@@ -0,0 +1,149 @@
+import { action } from '@ember/object';
+import isEmpty from 'lodash/isEmpty';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import isEmailValid from '../../utils/email-validator';
+import get from 'lodash/get';
+
+export default class LoginForm extends Component {
+  @service intl;
+  @service url;
+  @service store;
+  @service session;
+
+  @tracked errorMessage = null;
+  @tracked isErrorMessagePresent = false;
+  @tracked isLoading = false;
+  @tracked password = null;
+  @tracked email = null;
+  @tracked passwordValidationMessage = null;
+  @tracked emailValidationMessage = null;
+
+  ERROR_MESSAGES;
+
+  @action
+  async authenticate(event) {
+    event.preventDefault();
+    this.isLoading = true;
+    this._initErrorMessages();
+
+    const email = this.email;
+    const password = this.password;
+
+    if (!this.isFormValid) {
+      this.isLoading = false;
+      return;
+    }
+
+    if (this.args.isWithInvitation) {
+      try {
+        await this.args.certificationCenterInvitation.accept({
+          id: this.args.certificationCenterInvitationId,
+          code: this.args.certificationCenterInvitationCode,
+          email,
+        });
+        return this._authenticate(password, email);
+      } catch (errorResponse) {
+        errorResponse.errors.forEach((error) => {
+          const invitationIsAlreadyAcceptedOrUserIsAlreadyMember = error.status === '412';
+          if (invitationIsAlreadyAcceptedOrUserIsAlreadyMember) {
+            return this._authenticate(password, email);
+          }
+          this.errorMessage = this._handleResponseError(errorResponse);
+          this.isErrorMessagePresent = true;
+          this.isLoading = false;
+        });
+      }
+    }
+  }
+
+  async _authenticate(password, email) {
+    const scope = 'pix-certif';
+    try {
+      await this.session.authenticate('authenticator:oauth2', email, password, scope);
+    } catch (errorResponse) {
+      const errors = get(errorResponse, 'responseJSON');
+      this.errorMessage = this._handleResponseError(errors);
+      this.isErrorMessagePresent = true;
+    } finally {
+      this.isLoading = false;
+    }
```

Comment:
Pourquoi ne pas directement gérer les erreurs à la ligne 47 ?

```suggestion
    const scope = 'pix-certif';
    await this.session.authenticate('authenticator:oauth2', email, password, scope);
```